### PR TITLE
Add plugin to Java API

### PR DESCRIPTION
**Pull request by mudathirmahgoub**

### CODE REVIEW

File: src/api/java/jni/solver.cpp [hunk 1]
```diff
@@ -978,6 +979,25 @@ Java_io_github_cvc5_Solver_declareOracleFun(JNIEnv* env,
...
+JNIEXPORT void JNICALL Java_io_github_cvc5_Solver_addPlugin(JNIEnv* env,
+                                                            jobject,
+                                                            jlong pointer,
+                                                            jobject plugin)
+{
+  CVC5_JAVA_API_TRY_CATCH_BEGIN;
+  Solver* solver = reinterpret_cast<Solver*>(pointer);
+  jobject pluginReference = env->NewGlobalRef(plugin);
+  globalReferences[pointer].push_back(pluginReference);
+  ApiPlugin* p = new ApiPlugin(solver->getTermManager(), env, pluginReference);
```
Comment (mudathirmahgoub) [inline src/api/java/jni/solver.cpp:996]: I have two questions in this line:
- Is it ok, to use term manager associated with the solver? I am not sure why Plugin needs a term manager in its constructor @ajreynol .
- @aniemetz  we have to use a pointer here for the plugin and there is a memory leak here. Any suggestion for deleting this pointer other than global variable.

**Comment by ajreynol** [comment]
I don't think its strictly necessary to pass TermManager to the base class, although nearly all derived classes of this will need to take a TermManager, e.g. to construct lemmas.
If you have a strong opinion, I would maybe to open to removing it.

**Comment by mudathirmahgoub** [comment]
> I don't think its strictly necessary to pass TermManager to the base class, although nearly all derived classes of this will need to take a TermManager, e.g. to construct lemmas. If you have a strong opinion, I would maybe to open to removing it.

I agree. It makes more sense to add it to the base class.

**Comment by daniel-larraz** [comment]
This PR may need to be modified depending on how we decide to address coverage for the C++ API (see #10859)

**Review by aniemetz — APPROVED** [review]
LGTM

**Pull request merged**